onnxruntime/core/platform/windows/env.cc

```diff
@@ -76,9 +77,26 @@ class WindowsThread : public EnvThread {
   typedef HRESULT(WINAPI* SetThreadDescriptionFunc)(HANDLE hThread, PCWSTR lpThreadDescription);
   static unsigned __stdcall ThreadMain(void* param) {
     std::unique_ptr<Param> p((Param*)param);
-    // TODO: should I try to use SetThreadSelectedCpuSets?
-    if (!p->thread_options.affinity.empty())
-      SetThreadAffinityMask(GetCurrentThread(), p->thread_options.affinity[p->index]);
+    const KAFFINITY* affinities = reinterpret_cast<const KAFFINITY*>(p->thread_options.affinity.data());
+    size_t num_affinity = p->thread_options.affinity.size() * sizeof(size_t) / sizeof(KAFFINITY);
+    size_t offset = static_cast<size_t>(p->index) << 1;
+    if (offset + 1 < num_affinity) {
+      GROUP_AFFINITY thread_affinity;
+      memset(&thread_affinity, 0x0, sizeof(GROUP_AFFINITY));
+      thread_affinity.Group = static_cast<WORD>(affinities[offset]);
+      thread_affinity.Mask = affinities[offset + 1];
+      if (SetThreadGroupAffinity(GetCurrentThread(), &thread_affinity, nullptr)) {
+        LOGS_DEFAULT(WARNING) << "Set group affinity for sub-thread " << p->index << ", "
```

Comment:
what is a sub-thread? it's just a "thread". did you have anything else in mind?

Reply:
Renamed to "thread".